### ISSUE TITLE
Download to pause then resume when wifi cuts out then reconnects

### ISSUE THREAD

**Issue by SvyraCosmix** [Bionus/imgbrd-grabber#2445]
Just as the title says when internet is disconnected the download stops and does this (see image) cancels all the remaining files to be downloaded, wouldn't it make more sense for the downloading to pause then when the connection is reestablished the downloading would resume 

![Screenshot 2021-08-12 065140](https://user-images.githubusercontent.com/19334444/129209107-0a03762d-7dbb-4317-9257-1e64cccaeefd.png)

**Comment by Bionus** [comment]
> wouldn't it make more sense for the downloading to pause

Yes, I guess that's possible to detect common internet loss errors and pause the download rather than just keep failing.

> then when the connection is reestablished the downloading would resume

That's thougher. All Qt classes in 5.15 that provide this kind of stuff are deprecated and don't seem to work very well.
Qt 6 has [QNetworkInformation](https://doc-snapshots.qt.io/qt6-dev/qnetworkinformation.html) but Grabber still uses Qt 5.

**Comment by yami-no-tusbas** [comment]
Well, can't grabber like "retry" every half-hour or something in that case ? Like "network error detected" put the work in pause, start a counter for 30minutes, then try just to get a page from the job in waiting, if ti work, restart, else pause another time ?

**Comment by Bionus** [comment]
I guess it could, but that feels very hacky 🤔

**Comment by SvyraCosmix** [comment]
> I guess it could, but that feels very hacky 🤔

Good thing you didnt make the program to hack peoples computers, lmao in any case once I get another router it wont be a problem. It's not a huge problem just a suggestion

**Comment by yami-no-tusbas** [comment]
Well hacky, can't this function work ? it seems to be for an older QT : https://doc.qt.io/qt-5/qhostinfo.html
If the name lookup fail, we can assume the connection has a problem, especially if it was working before.
(note : I'm not a qt or c++ dev, i just looked the doc and it seems to do what we need) This seems to look for a route to a website, if this fail but the website was reachable before, then the connection is failing, pause, then retry later.
This would solve #1706 too, it's basicly the same.

**Comment by yami-no-tusbas** [comment]
Ok i'm testing it with a download list, disconnecting internet when t he job run.

1. try, the app crash here is the logs : [main.log](https://github.com/Bionus/imgbrd-grabber/files/7038805/main.log)
2. retry : The app stopped with an error, didn't reconnect the cable, just press resume, and crash : [main.log](https://github.com/Bionus/imgbrd-grabber/files/7038817/main.log)
So for the moment 6fb284b4 seems to crash when an error happen.

Now with monitors : 
I used start now to force a refresh, then cut internet while running the list and... Well it skip them : 

```
[13:41:45.681][Info] Monitoring new images for 'precure' on 'Gelbooru'
[13:41:45.692][Info] [gelbooru.com][Xml] Loading page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=0&amp;tags=precure`
[13:41:45.730][Info] [gelbooru.com][Xml] Receiving page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=0&amp;tags=precure`
[13:41:45.730][Error] [gelbooru.com][Xml] Loading error: Network unreachable (99)
[13:41:45.731][Warning] [gelbooru.com] Loading using Xml failed. Retry using Html.
[13:41:45.731][Info] [gelbooru.com][Html] Loading page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=precure&amp;pid=0`
[13:41:45.761][Info] [gelbooru.com][Html] Receiving page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=precure&amp;pid=0`
[13:41:45.761][Error] [gelbooru.com][Html] Loading error: Network unreachable (99)
[13:41:45.762][Warning] [gelbooru.com] No valid source of the site returned result.
[13:41:45.762][Warning] No results for monitor 'precure' on site 'Gelbooru'
```

But at least it doesn't freeze at all, nor crashes.

**Comment by Bionus** [comment]
Strange, I did the very same test and it just kept giving me an error each time I clicked "resume" when I was without internet, no crash. Maybe it's something about simultaneous downloads?

> Well it skip them

That's expected, since monitors shouldn't block anything.
As long as the "last state" is correct it should be fine I believe.

**Comment by yami-no-tusbas** [comment]
So my steps : 
use gelbooru as source, download first page (20 images), open the newtork interface window ready to deactivate the network adaptater. I set my simultaneous to one for the experience.
Launch the download, error happens, I can resume or cancel, I choos resume. Error looping, can't continue (witch is fine, network is down atm)
Reconnect the newtork, press resume, and its fine. Hum 🤔 

Trying with 10 simultaneous download now. (I changed the limit to 200 images because 10 by 10 it goes fast)
same procedure.
Stoping it in the midle of a file, resuming and no crash too. It restart fine.

That's strange, I can't reproduce it this time.
